managers/int.py

```diff
@@ -31,6 +37,162 @@ def __init__(self, controller: Controller) -> None:
+                )
+                self.unis_src[uni_a.id] = src_a.id
+                try:
+                    pp = self.get_proxy_port_or_raise(uni_a.id, evc_id)
+                except ProxyPortDestNotFound:
+                    pp = self.srcs_pp[src_a.id]
+                pp.evc_ids.add(evc_id)
+
+            if uni_z and "proxy_port" in uni_z.metadata:
+                src_z = uni_z.switch.get_interface_by_port_no(
+                    uni_z.metadata["proxy_port"]
+                )
+                self.unis_src[uni_z.id] = src_z.id
+                try:
+                    pp = self.get_proxy_port_or_raise(uni_z.id, evc_id)
+                except ProxyPortDestNotFound:
+                    pp = self.srcs_pp[src_z.id]
+                pp.evc_ids.add(evc_id)
+
+    async def handle_pp_link_down(self, link: Link) -> None:
+        """Handle proxy_port link_down."""
+        if not settings.FALLBACK_TO_MEF_LOOP_DOWN:
+            return
+        pp = self.srcs_pp.get(link.endpoint_a.id)
+        if not pp:
+            pp = self.srcs_pp.get(link.endpoint_b.id)
+        if not pp or not pp.evc_ids:
+            return
+
+        async with self._topo_link_lock:
+            evcs = await api.get_evcs(
+                **{
+                    "metadata.telemetry.enabled": "true",
+                    "metadata.telemetry.status": "UP",
+                }
+            )
+            to_deactivate = {
+                evc_id: evc for evc_id, evc in evcs.items() if evc_id in pp.evc_ids
+            }
+            if not to_deactivate:
+                return
+
+            log.info(
+                f"Handling link_down {link}, removing INT flows falling back to "
+                f"mef_eline, EVC ids: {list(to_deactivate)}"
+            )
+            metadata = {
+                "telemetry": {
+                    "enabled": True,
+                    "status": "DOWN",
+                    "status_reason": ["proxy_port_down"],
+                    "status_updated_at": datetime.utcnow().strftime(
+                        "%Y-%m-%dT%H:%M:%S"
+                    ),
+                }
+            }
+            await self.remove_int_flows(to_deactivate, metadata)
+
+    async def handle_pp_link_up(self, link: Link) -> None:
+        """Handle proxy_port link_up."""
+        if not settings.FALLBACK_TO_MEF_LOOP_DOWN:
+            return
+        pp = self.srcs_pp.get(link.endpoint_a.id)
+        if not pp:
+            pp = self.srcs_pp.get(link.endpoint_b.id)
+        if not pp or not pp.evc_ids:
+            return
+
+        # This sleep is to wait for at least a few seconds to ensure that the other
+        # proxy port would also have been considered up since the request
+        # on topology setting metadata might end up delaying, check out issue
+        # https://github.com/kytos-ng/of_lldp/issues/100
+        # TODO this will be optimized later on before releasing this NApp
+        await asyncio.sleep(5)
```

Comment:
I didnt understand this wait here. Can you please elaborate more about it?

Reply:
@italovalcy, this sleep will get removed once https://github.com/kytos-ng/of_lldp/issues/100, is addressed. But, yes, let me elaborate on it:

To make sure that indeed [both proxy ports are UP - for bidirectional EVCs - ](https://github.com/kytos-ng/telemetry_int/blob/feat/handle_proxy_port_link/managers/int.py#L163-L164) we're also relying on the `loop` state that's set via metadata, which `of_lldp` will set via a request, so there's a possibility that as highlighted in the linked issue, that this request is being made on a [of_lldp handler](https://github.com/kytos-ng/of_lldp/blob/master/main.py#L186-L192), which will get executed in the `app` threadpool at the moment, then that thread might get delayed in a extreme worst case scenario, and then maybe the metadata won't be there yet, so proxy ports wouldn't be considered up, and consequently not safe to assume that flows could get installed. In that issue, `of_lldp` will set the metadata upfront in memory, eliminating this wait here to ensure the request has suceeded, so this sleep is also assuming that 5 secs would be a worst case in the mean time, which should be OK, but far from ideal, that's that rationale.

Reply:
btw, there's also another one that's desirable to have when it comes to enhancing loop detection robustness, currently they're susceptible to starvation, this part it'll be handled on another issue https://github.com/kytos-ng/of_lldp/issues/97